source/develop/release-management/features/virt/vm-screenshot.md

```diff
@@ -0,0 +1,83 @@
+---
+title: VM Screenshot
+category: feature
+authors: sabusale
+---
+
+# VM Screenshot
+
+## Summary
+
+Allow users to take screenshot of the VM
+
+## Owner
+
+*   Name: Saif Abu Saleh (sabusale)
+*   Email: sabusale@redhat.com
+
+## Description
+
+User can take a screenshot of running VM
+
+### UI
+
+Not supported
+
+### Backend
+
+Backend will call VDSM to capture screenshot and will receive the `ppm`[[1]] file content
+and then `ppm` file content will be returned
+
+### VDSM
+
+VDSM will call Libvirt API[[2]] that capture the screenshot and saves it on given path on the host file system
+and then will return the `ppm` file content and remove the image on the host file system
```

Comment:
it's a quick operation, right?

Reply:
actually no, since the file is not compressed, see: https://en.wikipedia.org/wiki/Netpbm#:~:text=The%20PPM%20format%20is%20not,file%20size%20is%2073%2C848%20bytes.

I took image and it was with size 2.3MB, so it will take time to read the file and return it

Reply:
maybe it can cause an issue to the API?

lets assume the image size is >5Mb
download speed is less then 1Mbit/s

the response from VDSM alone will take > 1 minute (only the download part)
not considering other things like running the command, parse the file, ....

Reply:
we typically don't want clients to wait on long operations and so we implement them as asynchronous operations but it would really complicate the implementation here

so ok, if it can take relatively long time and we implement it as a synchronous operation, I think we'll need to make it crystal clear that the current implementation is intended for debugging also at the API level